ReClass.NET/Forms/SettingsForm.cs

```diff
@@ -149,5 +152,20 @@ private void SetTypedefinitionBindings()
 			SetBinding(utf16TextTypeTextBox, nameof(TextBox.Text), typeMapping, nameof(CppTypeMapping.TypeUtf16Text));
 			SetBinding(functionPtrTypeTextBox, nameof(TextBox.Text), typeMapping, nameof(CppTypeMapping.TypeFunctionPtr));
 		}
+
+		private void runAsAdminCheckBox_CheckedChanged(object sender, EventArgs e)
+		{
+			settings.RunAsAdmin = runAsAdminCheckBox.Checked;
+
+			if (runAsAdminCheckBox.Checked && !WinUtil.IsAdministrator)
+			{
+				var msgboxResult = MessageBox.Show("Would you like to restart ReClass.NET as administrator now?", "ReClass.NET", MessageBoxButtons.YesNo, MessageBoxIcon.Information);
```

Comment:
Maybe just notify the user that he has to restart ReClass to apply the change. This would remove the logic code from the settings form.